### PR TITLE
Z3 Optimize: add `assert_and_track` and `get_unsat_core`

### PR DESCRIPTION
**Pull request by toolCHAINZ**
I noticed these are missing from `Optimize` and would have some use for them.

### CODE REVIEW

File: z3/src/optimize.rs [hunk 1]
```diff
@@ -105,6 +126,43 @@ impl<'ctx> Optimize<'ctx> {
         unsafe { Z3_optimize_minimize(self.ctx.z3_ctx, self.z3_opt, ast.get_z3_ast()) };
     }
 
+    /// Return a subset of the assumptions provided to either the last
+    ///
+    /// * [`Optimize::check`] call.
```
Comment (waywardmonkeys) [inline z3/src/optimize.rs:131]: This seems like it needs editing?

**Review by waywardmonkeys — CHANGES_REQUESTED** [review]
Looks good to me apart from the comment… would it be easy to add a test for this?

File: z3/src/optimize.rs (hunk 1 above)
Comment (toolCHAINZ) [reply z3/src/optimize.rs:131]: Ooops, I'll fix that!

**Comment by toolCHAINZ** [comment]
> Looks good to me apart from the comment… would it be easy to add a test for this?

Ok, I've added a test that exercises both `assert_and_track + check` and an individual `check`.

**Comment by waywardmonkeys** [comment]
> Ok, I've added a test that exercises both `assert_and_track + check` and an individual `check`.

Great, thanks!

**Pull request merged**